src/Application/Actions/Hooks/StripePaymentsUpdate.php

```diff
@@ -140,41 +139,8 @@ private function handleChargeSucceeded(Event $event, Response $response): Respon
         // For now we support the happy success path –
         // as this is the only event type we're handling right now besides refunds.
         if ($charge->status === 'succeeded') {
-            /**
-             * @var array|Card|null $card
-             */
-            $card = $charge->payment_method_details?->toArray()['card'] ?? null;
-            if (is_array($card)) {
-                /** @var Card $card */
-                $card = (object)$card;
-            }
-
-            $cardBrand = CardBrand::fromNameOrNull($card?->brand);
-            $cardCountry = Country::fromAlpha2OrNull($card?->country);
-            $balanceTransaction = (string) $charge->balance_transaction;
-
-            // To give *simulated* webhooks, for Donation API-only load tests, an easy way to complete
-            // without crashing, we support skipping the original fee derivation by omitting
-            // `balance_transaction`. Real stripe charge.succeeded webhooks should always have
-            // an associated Balance Transaction.
-            if (!empty($balanceTransaction)) {
-                $originalFeeFractional = $this->getOriginalFeeFractional(
-                    $balanceTransaction,
-                    $donation->getCurrencyCode(),
-                );
-            } else {
-                $originalFeeFractional = $donation->getOriginalPspFee();
-            }
-
-            $donation->collectFromStripeCharge(
-                chargeId: $charge->id,
-                totalPaidFractional: $charge->amount,
-                transferId: (string)$charge->transfer,
-                cardBrand: $cardBrand,
-                cardCountry: $cardCountry,
-                originalFeeFractional: (string)$originalFeeFractional,
-                chargeCreationTimestamp: $charge->created,
-            );
+            $donationService = $this->donationService;
+            $donationService->updateDonationStatusFromSucessfulCharge($charge, $donation);
```

Comment:
code moved into donation service essentially unchanged to allow for re-use within the synchronous mandate setup processes